File: ed2/finale/annoluce/att/annoluce.c
#include <stdio.h>
#include <assert.h>

void mappatura(int N, int X[], int Y[], int Z[]) {
    // Mettete qui il codice della soluzione
}

int query(int D) {
    // Mettete qui il codice della soluzione
    return 42;
}


#define MAXN 100000
int X[MAXN], Y[MAXN], Z[MAXN];

int main() {
    FILE *fr, *fw;
    int N, Q, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(1 == fscanf(fr, "%d", &N));
    for(i=0; i<N; i++)
        assert(3 == fscanf(fr, "%d%d%d", &X[i], &Y[i], &Z[i]));

    mappatura(N, X, Y, Z);

    assert(1 == fscanf(fr, "%d", &Q));
    for(i=0; i<Q; i++) {
        int D;
        assert(1 == fscanf(fr, "%d", &D));
        fprintf(fw, "%d\n", query(D));
    }

    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/finale/annoluce/att/annoluce.pas
const
    MAXN = 100000;

(* Dichiarate qui eventuali variabili globali *)

procedure mappatura(N: longint; var X, Y, Z: array of longint);
begin
    (* Mettete qui il codice della soluzione *)
end;

function query(D: longint) : longint;
begin
    (* Mettete qui il codice della soluzione *)
    query := 42;
end;


var
    N, Q, D, i : longint;
    X, Y, Z    : array[0..MAXN-1] of longint;
    fr, fw : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);

    readln(fr, N);
    for i:=0 to N-1 do
        readln(fr, X[i], Y[i], Z[i]);

    mappatura(N, X, Y, Z);

    readln(fr, Q);
    for i:=0 to Q-1 do
    begin
        readln(fr, D);
        writeln(fw, query(D));
    end;

    close(fr);
    close(fw);
end.
